### PR TITLE
iOS Style Dictionary

### PR DESCRIPTION
**Pull request by mohsen-vend**
Create a separate process to generate custom files for iOS

### CODE REVIEW

File: lib/ios/format.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    const uiColorBlock = dictionary.allProperties
+      .map((token) => {
+        const nameParts = token.path.filter(
+          (part) => part.toLowerCase() !== "color"
+        );
+        const name = nameParts.join("").replace(/-/g, ""); // Remove dashes and combine
+
+        return `    static let ${name} = UIColor(hex: "${token.value}")`;
+      })
+      .join("\n");
```
Comment (BalbinaK) [inline lib/ios/format.js:36]: Seems like you're doing the exact same filtering in `swiftUIColorBlock` and `uiColorBlock`. Do they need to be separate constants?

Comment (BalbinaK) [reply lib/ios/format.js:36]: Ah, I see now that the difference lies in `Color(hex:....` vs. `UIColor(hex:...` 🔎

File: lib/ios/format.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+    // Generate SwiftUI colors
+    const swiftUIColorBlock = dictionary.allProperties
+      .filter((token) => !token.path[0].startsWith("color")) // Exclude primitive colors, only keep component colors
+      .filter((token) => !token.path[1].startsWith("color")) // Exclude semantic tokens, only keep component colors
+      .map((token) => {
+        let name = toCamelCase(token.path.slice(1).join("-")).replace(
+          "Default",
+          ""
+        ); // Convert to camelCase, removing first two parts
```
Comment (BalbinaK) [inline lib/ios/format.js]: I guess it's the first part that's removed here (`slice(1)`)?

File: processIos.js [hunk 3]
```diff
@@ -0,0 +1,674 @@
...
+      `${brand}-light`,
+      "semantic.json"
+    );
+    const colorFilePath = path.join(
+      tokensPath,
+      `${brand}-light`,
+      "colors.json"
+    ); // Load the color file for reference resolution
+
+    // Skip this brand if semantic.json or semantic.json or colors.json files are missing
```
Comment (BalbinaK) [inline processIos.js]:
```suggestion
    // Skip this brand if components.json or semantic.json or colors.json files are missing
```

**Review by BalbinaK — APPROVED** [review]
LGTM! 🎉

File: lib/ios/format.js (hunk 2 above)
Comment (mohsen-vend) [reply lib/ios/format.js]: nice catch 👍 there were two of these, changed both comments

**Pull request merged**